src/VisualStudio/CSharp/Impl/ProjectSystemShim/HACK_VariantStructure.cs

```diff
@@ -23,10 +23,10 @@ namespace Microsoft.VisualStudio.LanguageServices.CSharp.ProjectSystemShim
     internal struct HACK_VariantStructure
     {
         [FieldOffset(0)]
-        private short _type;
+        private readonly short _type;
 
         [FieldOffset(8)]
-        private short _booleanValue;
+        private readonly short _booleanValue;
```

Comment:
Note this is used in GetObjectForNativeVariant as called below, which means these fields *are* mutated. The refactoring is probably incorrect here.

Reply:
(I presume the Marshal call will happily overwrite the values and ignore 'readonly'. That probably warrants a #pragma suppress.)

Reply:
➡️ This goes the other way around. `GetObjectForNativeVariant` is reading from these fields, not writing to them.